lapack/tpls/KokkosLapack_svd_tpl_spec_decl.hpp

```diff
@@ -0,0 +1,679 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSLAPACK_SVD_TPL_SPEC_DECL_HPP_
+#define KOKKOSLAPACK_SVD_TPL_SPEC_DECL_HPP_
+
+#include "KokkosKernels_Error.hpp"
+#include "Kokkos_ArithTraits.hpp"
+
+namespace KokkosLapack {
+namespace Impl {
+template <class ExecutionSpace, class AMatrix, class SVector, class UMatrix,
+          class VMatrix>
+inline void svd_print_specialization() {
+#ifdef KOKKOSKERNELS_ENABLE_CHECK_SPECIALIZATION
+#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSOLVER
+  if constexpr (std::is_same_v<ExecutionSpace, Kokkos::Cuda>) {
+    printf(
+        "KokkosLapack::svd<> TPL Cusolver specialization for < %s , %s, %s, %s "
+        ">\n",
+        typeid(AMatrix).name(), typeid(SVector).name(), typeid(UMatrix).name(),
+        typeid(VMatrix).name());
+  }
+#endif
+#endif
+}
+}  // namespace Impl
+}  // namespace KokkosLapack
+
+// LAPACK
+#ifdef KOKKOSKERNELS_ENABLE_TPL_LAPACK
+#include "KokkosLapack_Host_tpl.hpp"
+
+namespace KokkosLapack {
+namespace Impl {
+
+template <class ExecutionSpace, class AMatrix, class SVector, class UMatrix,
+          class VMatrix>
+void lapackSvdWrapper(const ExecutionSpace& /* space */, const char jobu[],
+                      const char jobvt[], const AMatrix& A, const SVector& S,
+                      const UMatrix& U, const VMatrix& Vt) {
+  using memory_space = typename AMatrix::memory_space;
+  using Scalar       = typename AMatrix::non_const_value_type;
+  using Magnitude    = typename SVector::non_const_value_type;
+  using ALayout_t    = typename AMatrix::array_layout;
```

Comment:
Could probably just static_assert that the layouts are all LayoutLeft, and then simplify the lda/ldu/ldvt below